src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -1241,7 +1241,7 @@ def get_text_position_embeddings(
             )
         if text_position_ids is not None:
             text_position_embedding = get_sine_pos_embed(
-                text_position_ids[..., None], num_pos_feats=self.d_model, exchange_xy=False
+                text_position_ids[..., None].to(text_features.dtype), num_pos_feats=self.d_model, exchange_xy=False
```

Comment:
> Is this change needed? If position_embeddings is created in the correct dtype then we shouldn't need to do this, I think

Thanks for your comment. Furthing tracing the code, it is confirmed that the dtype of `text_position_embedding` does not align with `text_features`.  I recast `text_position_ids` here since I do not want to change the original data type.